wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm

```diff
@@ -0,0 +1,151 @@
+#import "ZXing/ImageView.h"
+#import "ZXing/Result.h"
+#import "ZXIFormatHelper.h"
+#import "ZXIPosition+Helper.h"
+
+using namespace ZXing;
+
+@interface ZXIBarcodeReader()
+@property (nonatomic, strong) CIContext* ciContext;
+@end
+
+@implementation ZXIBarcodeReader
+
+- (instancetype)init {
+    return [self initWithHints: [[ZXIDecodeHints alloc]initWithTryHarder:NO tryRotate:NO tryDownscale:NO maxNumberOfSymbols:1 formats:@[]]];
+}
+
+- (instancetype)initWithHints:(ZXIDecodeHints*)hints{
+    self = [super init];
+    self.ciContext = [CIContext new];
+    self.hints = hints;
+    return self;
+}
+
+- (NSArray<ZXIResult *> *)readCVPixelBuffer:(nonnull CVPixelBufferRef)pixelBuffer {
+    OSType pixelFormat = CVPixelBufferGetPixelFormatType(pixelBuffer);
+
+    // We tried to work with all luminance based formats listed in kCVPixelFormatType
+    // but only the following ones seem to be supported on iOS.
+    switch (pixelFormat) {
+        case kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange:
+        case kCVPixelFormatType_420YpCbCr8BiPlanarFullRange:
+            NSInteger cols = CVPixelBufferGetWidth(pixelBuffer);
+            NSInteger rows = CVPixelBufferGetHeight(pixelBuffer);
+            NSInteger bytesPerRow = CVPixelBufferGetBytesPerRowOfPlane(pixelBuffer, 0);
+            CVPixelBufferLockBaseAddress(pixelBuffer, kCVPixelBufferLock_ReadOnly);
+            const uint8_t * bytes = static_cast<const uint8_t *>(CVPixelBufferGetBaseAddressOfPlane(pixelBuffer, 0));
+            ImageView imageView = ImageView(
+                                            static_cast<const uint8_t *>(bytes),
+                                            static_cast<int>(cols),
+                                            static_cast<int>(rows),
+                                            ImageFormat::Lum,
+                                            static_cast<int>(bytesPerRow),
+                                            0);
+            NSArray* results = [self readImageView:imageView];
+            CVPixelBufferUnlockBaseAddress(pixelBuffer, kCVPixelBufferLock_ReadOnly);
+            return results;
+    }
+
+    // If given pixel format is not a supported type with a luminance channel we just use the
+    // default method
+    return [self readCIImage:[[CIImage alloc] initWithCVImageBuffer:pixelBuffer]];
+}
+
+- (NSArray<ZXIResult *> *)readCIImage:(nonnull CIImage *)image {
+    CGImageRef cgImage = [self.ciContext createCGImage:image fromRect:image.extent];
+    auto results = [self readCGImage:cgImage];
+    CGImageRelease(cgImage);
+    return results;
+}
+
+- (NSArray<ZXIResult*> *)readImageView: (ImageView)imageView {
+    Results results = ReadBarcodes(imageView, [ZXIBarcodeReader DecodeHintsFromZXIOptions:self.hints]);
+
+    NSMutableArray* zxiResults = [NSMutableArray array];
+    for (auto result: results) {
+        if(result.status() == DecodeStatus::NoError) {
+            const std::wstring &resultText = result.text();
```

Comment:
Please merge the latest master changes and add the `#define ZX_USE_UTF8` before including `ReadBarcode.h` or in your project settings. See [here](https://github.com/nu-book/zxing-cpp/issues/338#issuecomment-1160989296) for background info. `text()` will return a utf8 encoded `std::string` then (and only that moving forward).